packages/core/src/components/carousel/carousel.tsx

```diff
@@ -1,142 +1,201 @@
+
+  componentWillLoad() {
+    this.carouselItems = this.element.querySelectorAll('atom-carousel-item')
+
+    if (this.carouselItems?.length <= 1) {
+      this.loop = false
+      this.hasNavigation = false
+      this.hasPagination = false
+    }
+  }
 
   componentDidLoad() {
-    this.swiperEl = this.host.querySelector('swiper-container')
-
-    const children = this.host.querySelectorAll('atom-carousel-item')
-
-    children.forEach((child) => {
-      const slide = child.querySelector('swiper-slide')
-
-      if (slide) {
-        this.swiperEl.appendChild(slide)
-        this.host.removeChild(child)
-      }
-    })
-
-    this.swiperEl.swiper?.on('slideChange', () => {
-      this.atomChange.emit()
-    })
-    this.swiperEl.swiper?.on('navigationPrev', () => {
-      this.atomClickPrev.emit()
-    })
-    this.swiperEl.swiper?.on('navigationNext', () => {
-      this.atomClickNext.emit()
-    })
-
-    let params: { pagination: PaginationOptions } = this.pagination && {
-      pagination: {
-        type:
-          this.paginationType === 'thumbnails' ? 'custom' : this.paginationType,
-        clickable: this.paginationClickable,
-      },
+    this.carouselWrapper =
+      this.element.shadowRoot.querySelector('.carousel-wrapper')
+    this.nextButton = this.element.shadowRoot.querySelector('.navigation--next')
+    this.prevButton = this.element.shadowRoot.querySelector('.navigation--prev')
+
+    this.carouselWrapper.addEventListener('touchstart', this.handleTouchStart)
+    this.carouselWrapper.addEventListener('touchend', this.handleTouchEnd)
+
+    this.showOrHideNavigationButtons()
+
+    if (this.autoplay > 0) this.startAutoplay()
+  }
+
+  disconnectedCallback() {
+    this.carouselWrapper.removeEventListener(
+      'touchstart',
+      this.handleTouchStart
+    )
+    this.carouselWrapper.removeEventListener('touchend', this.handleTouchEnd)
+  }
+
+  handleMoveToPaginationItem(index: number) {
+    if (!this.carouselItems || !this.carouselItems[index]) return
+
+    this.currentIndex = index
+    this.showOrHideNavigationButtons()
+  }
+
+  handleNavigationClick(event: Event) {
+    const button = event.currentTarget as HTMLButtonElement
+    const isNext = button?.classList.contains('navigation--next')
+    let newIndex = this.currentIndex + (isNext ? 1 : -1)
+
+    const isLastItemAndNext =
+      isNext && this.currentIndex === this.carouselItems.length - 1
+    const isFirstItemAndPrev = !isNext && this.currentIndex === 0
+
+    if (!this.loop && (isLastItemAndNext || isFirstItemAndPrev)) return
+
+    if (this.loop) {
+      newIndex =
+        (newIndex + this.carouselItems.length) % this.carouselItems.length
     }
 
-    if (this.paginationType === 'thumbnails') {
-      const urls = this.thumbnailImages ? this.thumbnailImages.split(',') : []
+    this.currentIndex = newIndex
+    this.showOrHideNavigationButtons()
+  }
+
+  showOrHideNavigationButtons() {
+    if (!this.nextButton || !this.prevButton) return
```

Comment:
Wouldn't it be the same thing  to use `this.hasNavigation`?

Reply:
No because `!this.nextButton || !this.prevButton` checks the `querySelector` of buttons and `this.hasNavigation` is a prop, but I improved this function here [fe89528](https://github.com/juntossomosmais/atomium/pull/434/commits/fe89528fdb8c8053931afef27f8e9a9c0f78ba07)

Reply:
if `hasNavegation` is `false`, the `querySelector` will be `undefined`, right?

Reply:
Yes! But if `hasNavegation` is `true` and happens some error to load `button` `querySelector` will be `undefined` too and create an execution error, so is better to validate all scenarios